File: XCase.Swagger.ProxyGenerator/RAML/YamlException.cs
namespace XCase.REST.ProxyGenerator.RAML
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.Threading.Tasks;

    public class YamlException : Exception
    {
        /// <summary>
        /// Gets the position in the input stream where the event that originated the exception starts.
        /// </summary>
        public Mark Start { get; private set; }

        /// <summary>
        /// Gets the position in the input stream where the event that originated the exception ends.
        /// </summary>
        public Mark End { get; private set; }

        /// <summary>
        /// Initializes a new instance of the <see cref="YamlException"/> class.
        /// </summary>
        public YamlException()
        {
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="YamlException"/> class.
        /// </summary>
        /// <param name="message">The message.</param>
        public YamlException(string message)
            : base(message)
        {
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="YamlException"/> class.
        /// </summary>
        public YamlException(Mark start, Mark end, string message)
            : this(start, end, message, null)
        {
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="YamlException"/> class.
        /// </summary>
        public YamlException(Mark start, Mark end, string message, Exception innerException)
            : base(string.Format("({0}) - ({1}): {2}", start, end, message), innerException)
        {
            Start = start;
            End = end;
        }
        /// <summary>
        /// Initializes a new instance of the <see cref="YamlException"/> class.
        /// </summary>
        /// <param name="message">The message.</param>
        /// <param name="inner">The inner.</param>
        public YamlException(string message, Exception inner)
            : base(message, inner)
        {
        }
    }
}

File: platform/src/java/com/xcase/rest/generator/raml/YamlException.java
package com.xcase.rest.generator.raml;

import com.xcase.rest.generator.*;
import com.google.gson.*;
import com.xcase.common.impl.simple.core.CommonHTTPManager;
import com.xcase.common.utils.ConverterUtils;
import java.lang.invoke.MethodHandles;
import java.io.*;
import java.net.*;
import java.nio.file.*;
import java.util.*;
import org.apache.http.HttpEntity;
import org.apache.http.HttpResponse;
import org.apache.http.NameValuePair;
import org.apache.http.message.BasicNameValuePair;
import org.apache.http.util.EntityUtils;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class YamlException extends Exception {
    /// <summary>
    /// Gets the position in the input stream where the event that originated the exception starts.
    /// </summary>
    public Mark Start;

    /// <summary>
    /// Gets the position in the input stream where the event that originated the exception ends.
    /// </summary>
    public Mark End;

    /// <summary>
    /// Initializes a new instance of the <see cref="YamlException"/> class.
    /// </summary>
    public YamlException()
    {
        
    }

    /// <summary>
    /// Initializes a new instance of the <see cref="YamlException"/> class.
    /// </summary>
    /// <param name="message">The message.</param>
    public YamlException(String message) {
        super(message);
    }

    /// <summary>
    /// Initializes a new instance of the <see cref="YamlException"/> class.
    /// </summary>
    public YamlException(Mark start, Mark end, String message) {
        this(start, end, message, null);
    }

    /// <summary>
    /// Initializes a new instance of the <see cref="YamlException"/> class.
    /// </summary>
    public YamlException(Mark start, Mark end, String message, Exception innerException) {
        super("(" + start + ") - (" + end + "): " + message, innerException);
        Start = start;
        End = end;
    }
    /// <summary>
    /// Initializes a new instance of the <see cref="YamlException"/> class.
    /// </summary>
    /// <param name="message">The message.</param>
    /// <param name="inner">The inner.</param>
    public YamlException(String message, Exception innerException) {
        super(message, innerException);
    }
}
